### ISSUE TITLE
tokenizer of bert-base-uncased gives an incorrect split

### ISSUE THREAD

**Issue by kangxin** [huggingface/transformers#2220]
I found the following code gives an incorrect split

`tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')`
`text = "[CLS] Who was Jim Henson ? [SEP] Jim Henson was a puppeteer [SEP]"`
`tokenized_text = tokenizer.tokenize(text)`
`print(' '.join(tokenized_text))`

> [ cl ##s ] who was jim henson ? [ sep ] jim henson was a puppet ##eer [ sep ]

The correct one, based on the Quick tour example, seems to be

> [CLS] who was jim henson ? [SEP] jim heson was a puppet ##eer [SEP]

**Comment by TheEdoardo93** [comment]
As you can see, I'm not able to reproduce your bug, therefore **in my environment it works as expected without bugs**. Here the source code I've used (the same as you):
```
Python 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import transformers
>>> from transformers import BertTokenizer
>>> tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
>>> text = "[CLS] Who was Jim Henson ? [SEP] Jim Henson was a puppeteer [SEP]"
>>> tokenized_text = tokenizer.tokenize(text)
>>> tokenized_text
['[CLS]', 'who', 'was', 'jim', 'henson', '?', '[SEP]', 'jim', 'henson', 'was', 'a', 'puppet', '##eer', '[SEP]']
>>> ' '.join(tokenized_text)
'[CLS] who was jim henson ? [SEP] jim henson was a puppet ##eer [SEP]' 
```

My environment specifications are the following:
- Python: **3.6.9**
- OS: **Ubuntu 16.04**
- Transformers: **2.2.2** (installed with `pip install --upgrade transformers`)
- PyTorch: **1.3.1**
- TensorFlow: **2.0**

Please specify your environment in order to understand why in your case it doesn't work.

> I found the following code gives an incorrect split
> 
> `tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')`
> `text = "[CLS] Who was Jim Henson ? [SEP] Jim Henson was a puppeteer [SEP]"`
> `tokenized_text = tokenizer.tokenize(text)`
> `print(' '.join(tokenized_text))`
> 
> > [ cl ##s ] who was jim henson ? [ sep ] jim henson was a puppet ##eer [ sep ]
> 
> The correct one, based on the Quick tour example, seems to be
> 
> > [CLS] who was jim henson ? [SEP] jim heson was a puppet ##eer [SEP]

**Comment by BramVanroy** [comment]
This is an issue that is fixed in 2.2.2 IIRC. It was present in earlier 2.2.x versions. If you update to the latest versions, it should be fixed.

**Comment by dirkgr** [comment]
I am actually seeing that the behavior is now non-deterministic:
```
$ python -c 'import transformers; print(len(transformers.BertTokenizer.from_pretrained("bert-base-uncased").tokenize("A, [MASK] AllenNLP sentence.")))'
8
$ python -c 'import transformers; print(len(transformers.BertTokenizer.from_pretrained("bert-base-uncased").tokenize("A, [MASK] AllenNLP sentence.")))'
8
$ python -c 'import transformers; print(len(transformers.BertTokenizer.from_pretrained("bert-base-uncased").tokenize("A, [MASK] AllenNLP sentence.")))'
10
```

**Comment by dirkgr** [comment]
I proposed a fix in #2232.
